sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/TracezDataAggregator.java

```diff
@@ -0,0 +1,295 @@
+
+    /** Stores finished successful requests of duration within the interval [100us, 1ms). */
+    MICROSx100_MILLIx1(TimeUnit.MICROSECONDS.toNanos(100), TimeUnit.MILLISECONDS.toNanos(1)),
+
+    /** Stores finished successful requests of duration within the interval [1ms, 10ms). */
+    MILLIx1_MILLIx10(TimeUnit.MILLISECONDS.toNanos(1), TimeUnit.MILLISECONDS.toNanos(10)),
+
+    /** Stores finished successful requests of duration within the interval [10ms, 100ms). */
+    MILLIx10_MILLIx100(TimeUnit.MILLISECONDS.toNanos(10), TimeUnit.MILLISECONDS.toNanos(100)),
+
+    /** Stores finished successful requests of duration within the interval [100ms, 1sec). */
+    MILLIx100_SECONDx1(TimeUnit.MILLISECONDS.toNanos(100), TimeUnit.SECONDS.toNanos(1)),
+
+    /** Stores finished successful requests of duration within the interval [1sec, 10sec). */
+    SECONDx1_SECONDx10(TimeUnit.SECONDS.toNanos(1), TimeUnit.SECONDS.toNanos(10)),
+
+    /** Stores finished successful requests of duration within the interval [10sec, 100sec). */
+    SECONDx10_SECONDx100(TimeUnit.SECONDS.toNanos(10), TimeUnit.SECONDS.toNanos(100)),
+
+    /** Stores finished successful requests of duration greater than or equal to 100sec. */
+    SECONDx100_MAX(TimeUnit.SECONDS.toNanos(100), Long.MAX_VALUE);
+
+    private final long latencyLowerBound;
+    private final long latencyUpperBound;
+
+    /**
+     * Constructs a {@code LatencyBoundaries} with the given boundaries and label.
+     *
+     * @param latencyLowerBound the latency lower bound of the bucket.
+     * @param latencyUpperBound the latency upper bound of the bucket.
+     */
+    LatencyBoundaries(long latencyLowerBound, long latencyUpperBound) {
+      this.latencyLowerBound = latencyLowerBound;
+      this.latencyUpperBound = latencyUpperBound;
+    }
+
+    /**
+     * Returns the latency lower bound of the bucket.
+     *
+     * @return the latency lower bound of the bucket.
+     */
+    public long getLatencyLowerBound() {
+      return latencyLowerBound;
+    }
+
+    /**
+     * Returns the latency upper bound of the bucket.
+     *
+     * @return the latency upper bound of the bucket.
+     */
+    public long getLatencyUpperBound() {
+      return latencyUpperBound;
+    }
+  }
+
+  /**
+   * Returns a Map of counts for the {@link io.opentelemetry.trace.Status#OK} spans within
+   * [lowerBound, upperBound) {@link io.opentelemetry.sdk.extensions.zpages.TracezDataAggregator}.
+   *
+   * @param lowerBound latency lower bound (inclusive)
+   * @param upperBound latency upper bound (exclusive)
+   * @return a Map of span counts for each span name within the bounds.
+   */
+  public Map<String, Integer> getSpanLatencyCounts(long lowerBound, long upperBound) {
+    Collection<ReadableSpan> allCompletedSpans = spanProcessor.getCompletedSpans();
+    Map<String, Integer> numSpansPerName = new HashMap<>();
+    for (ReadableSpan span : allCompletedSpans) {
+      if (span.toSpanData().getStatus().isOk()
```

Comment:
@jkwatson What do you think about adding `.getStatus` to `ReadableSpan` (I guess it should really have all the getters?). Going through `toSpanData` just to check if the span is ok seems off.

Reply:
That sounds totally reasonble to me.